src/ir_Corona.cpp

```diff
@@ -0,0 +1,578 @@
+  else
+    setBits(&remote_state[kCoronaAcSectionData0Pos],
+            kCoronaAcFanOffset, kCoronaAcFanSize, speed);
+}
+
+/// Change the powersave setting.
+/// @param[in] on true, the setting is on. false, the setting is off.
+void IRCoronaAc::setEcono(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionData0Pos], kCoronaAcPowerSaveOffset, on);
+}
+
+/// Get the value of the current powersave setting.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getEcono(void) {
+  return GETBIT8(remote_state[kCoronaAcSectionData0Pos],
+                 kCoronaAcPowerSaveOffset);
+}
+
+/// Convert a standard A/C Fan speed into its native fan speed.
+/// @param[in] speed The desired stdAc::fanspeed_t fan speed
+/// @return The given fan speed in native format
+uint8_t IRCoronaAc::convertFan(const stdAc::fanspeed_t speed) {
+  switch (speed) {
+    case stdAc::fanspeed_t::kMin:
+    case stdAc::fanspeed_t::kLow:    return kCoronaAcFanLow;
+    case stdAc::fanspeed_t::kMedium: return kCoronaAcFanMedium;
+    case stdAc::fanspeed_t::kHigh:
+    case stdAc::fanspeed_t::kMax:    return kCoronaAcFanHigh;
+    default:                         return kCoronaAcFanAuto;
+  }
+}
+
+/// Convert a native fan speed to it's common equivalent.
+/// @param[in] speed The desired native fan speed
+/// @return The given fan speed in stdAc::fanspeed_t format
+stdAc::fanspeed_t IRCoronaAc::toCommonFanSpeed(const uint8_t speed) {
+  switch (speed) {
+    case kCoronaAcFanHigh:    return stdAc::fanspeed_t::kHigh;
+    case kCoronaAcFanMedium:  return stdAc::fanspeed_t::kMedium;
+    case kCoronaAcFanLow:     return stdAc::fanspeed_t::kLow;
+    default:                  return stdAc::fanspeed_t::kAuto;
+  }
+}
+
+/// Set the Vertical Swing mode of the A/C.
+/// @param[in] on true, the setting is on. false, the setting is off.
+/// @note This is a button press, and not a state
+///       after sending it once you should turn it off
+void IRCoronaAc::setSwingV(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionData0Pos],
+         kCoronaAcSwingVToogleOffset, on);
+}
+
+/// Get the Vertical Swing mode of the A/C.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getSwingV(void) {
+  return GETBIT64(remote_state[kCoronaAcSectionData0Pos],
+                  kCoronaAcSwingVToogleOffset);
+}
+
+/// Set the Timer time
+/// @param[in] section index of section, used for offset.
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (non in range value is disable).
+///   Valid is from 1 minute to 12 hours
+void IRCoronaAc::_setTimer(const uint8_t section, const uint16_t nr_of_mins) {
+  // default to off
+  uint16_t hsecs = kCoronaAcTimerOff;
+  if (1 <= nr_of_mins && nr_of_mins <= kCoronaAcTimerMax)
+    hsecs = nr_of_mins * kCoronaAcTimerUnitsPerMin;
+
+  uint8_t pos = section * kCoronaAcSectionBytes;
+  remote_state[pos + kCoronaAcSectionData1Pos] = (hsecs & 0xff00) >> 8;
```

Comment:
The bit mask is unnecessary.
i.e.
`remote_state[pos + kCoronaAcSectionData1Pos] = hsecs >> 8;` should be the same. NFI if the optimiser would clean that up or not.

Reply:
Doesn't changing that make it less clear on what is going on? (and I thought the compiler would be more picky, or rather I would actually want it to be more picky)

Reply:
Nope. C/C++ assume you know what you are doing. i.e. Of course you can shoot your own foot. Just pull that trigger! ;-)

As we try to make this as compact (code/flash-wise) as possible/practical removing redundant code is worth it. Especially if someone who knows the language knows what it will do/is going on. But yeah, I get that your code is clearer/more readable. I used to do it myself too. Add a comment maybe?

Reply:
I would assume that the compiler would generate identical code for both ways, just that it would generate warnings if you cast something... I have seen plenty of times where gcc has been changed to throw new warnings for totally valid cases, and that has caused ipxe to have to add hacks to avoid the warnings..

Yes comment added.